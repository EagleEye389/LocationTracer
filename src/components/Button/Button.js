import React from 'react'

const ButtonControl = ({action,styler,label,Isvisible})=>{
 return (  Isvisible && 
           <button className={styler} onClick={action}>{label}</button>
        )
}

export default ButtonControl;