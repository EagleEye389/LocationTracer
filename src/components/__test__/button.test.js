import React from 'react';
import Button from '../Button/Button';
import {shallow} from 'enzyme';

describe('Button component is rendering properly',()=>{

    it('Button is visible', () => {
     const buttonControl = shallow(<Button  Isvisible={true}/>)
     expect(buttonControl.find('button').length).toBe(1);
    });
    
    it('Button is hidden', () => {
        const buttonControl = shallow(<Button  Isvisible={false}/>)
        expect(buttonControl.find('button').length).toBe(0);
    });

    it('Button showing text properly',()=>{
        let buttonText = 'test';
        const buttonControl = shallow(<Button  Isvisible={true} label={buttonText} />)
        expect(buttonControl.props().children).toBe(buttonText)
    })

    it('Button class is rendering correctly',()=>{        
        let buttonClass = 'test';        
        const buttonControl = shallow(<Button  Isvisible={true} styler={ buttonClass }/>)
        expect(buttonControl.hasClass(buttonClass)).toBe(true)

    })

    it('Button action is working properly',()=>{  
        let a = 5;      
        const test =()=> a++;    
        const buttonControl = shallow(<Button  Isvisible={true}  action={test}/>)
        buttonControl.find('button').simulate('click');
        expect(a).toBe(6)
    })
    
    it('Button snapshot test',()=>{
        const buttonControl = shallow(<Button  Isvisible={true}  action={test}/>)
        expect(buttonControl).toMatchSnapshot();

    })

})
