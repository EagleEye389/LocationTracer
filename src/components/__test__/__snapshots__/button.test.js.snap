// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component is rendering properly Button snapshot test 1`] = `ShallowWrapper {}`;
